File: app/src/main/java/com/example/tesonet/database/daos/ServerDao.java
package com.example.tesonet.database.daos;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.tesonet.database.models.Server;

import java.util.List;

@Dao
public interface ServerDao {

    @Insert
    void insert(Server server);

    @Query("DELETE FROM servers_table")
    void deleteAllServers();

    @Query("SELECT * FROM servers_table")
    List<Server> getAllServers();
}

File: app/src/main/java/com/example/androidparty/database/ServerDAO.kt
package com.example.androidparty.database

import androidx.room.*

@Dao
interface ServerDAO {
    @Query("SELECT * FROM servers")
    fun getServers(): List<ServerEntity>

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateServer(server: ServerEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addServer(server: ServerEntity)

    @Delete
    fun deleteServer(server: ServerEntity)
}